pipeline/authn/authenticator_oauth2_introspection.go

```diff
@@ -168,10 +168,6 @@ func (a *AuthenticatorOAuth2Introspection) Authenticate(r *http.Request, session
 			return errors.WithStack(err)
 		}
 
-		if len(i.TokenType) > 0 && i.TokenType != "access_token" {
-			return errors.WithStack(helper.ErrForbidden.WithReason(fmt.Sprintf("Introspected token is not an access token but \"%s\"", i.TokenType)))
-		}
```

Comment:
So ORY Hydra 1.9.0 introduces a new parameter `token_use` which can be `access_token` or `refresh_token`. I think we should keep this method here but add `TokenUse string                 `json:"token_use"`` [here](https://github.com/ory/oathkeeper/pull/556/files#diff-5e85f382e854a595d5161855726166f6R88).

It would be nice if you could also add a test for this!

Reply:
Got it! We will check for `token_use` instead of `token_type`. I'll update this PR.

Reply:
Could you check if is [it](https://github.com/ory/oathkeeper/pull/556/files#diff-5c8061290da17d352a4b888e5270cdd071ea584cd4751011132d8cccd69594f0R172-R174) what you were talking about?